tests/macros/apiref.test.js

```diff
@@ -0,0 +1,262 @@
+                localization_tags: [],
+                locale: 'ja',
+                last_edit: '2018-12-18T15:36:54.445468',
+                review_tags: []
+            }
+        ],
+        summary:
+            'The <code><strong>Gamepad.buttons</strong></code> property of the <a href="/en-US/docs/Web/API/Gamepad" title="The Gamepad interface of the Gamepad API defines an individual gamepad or other controller, allowing access to information such as button presses, axis positions, and id."><code>Gamepad</code></a> interface returns an array of <a href="/en-US/docs/Web/API/GamepadButton" title="The GamepadButton interface defines an individual button of a gamepad or other controller, allowing access to the current state of different types of buttons available on the control device."><code>gamepadButton</code></a> objects representing the buttons present on the device.',
+        id: 87763,
+        review_tags: [],
+        slug: 'Web/API/Gamepad/buttons',
+        uuid: '6b6f5f4d-81cb-4b65-81dc-88dc94bbd494',
+        title: 'Gamepad.buttons',
+        url: '/en-US/docs/Web/API/Gamepad/buttons',
+        modified: '2019-01-16T13:03:39.199620',
+        label: 'Gamepad.buttons',
+        localization_tags: [],
+        last_edit: '2018-07-20T12:21:09.854338',
+        sections: [
+            {
+                id: 'Quick_Links',
+                title: null
+            },
+            {
+                id: 'Syntax',
+                title: 'Syntax'
+            },
+            {
+                id: 'Example',
+                title: 'Example'
+            },
+            {
+                id: 'Value',
+                title: 'Value'
+            },
+            {
+                id: 'Specifications',
+                title: 'Specifications'
+            },
+            {
+                id: 'Browser_compatibility',
+                title: 'Browser compatibility'
+            },
+            {
+                id: 'sect1',
+                title: null
+            },
+            {
+                id: 'sect2',
+                title: null
+            },
+            {
+                id: 'sect3',
+                title: null
+            },
+            {
+                id: 'sect4',
+                title: null
+            },
+            {
+                id: 'sect5',
+                title: null
+            },
+            {
+                id: 'sect6',
+                title: null
+            },
+            {
+                id: 'Legend',
+                title: 'Legend'
+            },
+            {
+                id: 'See_also',
+                title: 'See also'
+            }
+        ]
+    }
+];
```

Comment:
In one of ExeBoss's PRs that has not landed yet (I think) there is a utility function to read fixtures like this giant JSON blob from files. If this subpagesExpand data could be useful for other macro tests, consider putting it in a fixture file and reading it in with a utility function

Reply:
The PR adding those functions is #1073, and I'd say it’s close to merging.

Reply:
I have extracted the `readJSONFixture(…)` helper function into #&lt;pr&gt;.